AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeGenerator.cs

```diff
@@ -139,6 +142,20 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
                 };
                 await Write(enumTemplate, Path.Combine("Models", enumTemplate.Model.TypeDefinitionName + ".cs"));
             }
+
+            // Parameter groups
+            foreach (Method method in serviceClient.Methods)
+            {
+                foreach (string parameterGroup in method.ParameterGroups)
+                {
+                    Parameter groupParameter = method.Parameters.First(p => string.Equals(p.Type.Name, parameterGroup, StringComparison.InvariantCultureIgnoreCase));
+                    var parameterGroupTemplate = new ParameterGroupTemplate
+                    {
+                        Model = new ParameterGroupTemplateModel(serviceClient, groupParameter)
+                    };
+                    await Write(parameterGroupTemplate, Path.Combine("Parameters", parameterGroupTemplate.Model.ParameterGroupType + ".cs"));
```

Comment:
Does that mean we will create the group parameter types under \\Model\\Parameters? It is just fine to produce under \\Models? Those are all model classes anyway.